src/services/item/plugins/invitation/service.ts

```diff
@@ -223,6 +224,34 @@ export class InvitationService {
     return { memberships, invitations };
   }
 
+  /**
+   * Create memberships and invitations given invitation list
+   * @param actor
+   * @param repositories
+   * @param itemId
+   * @param invitations
+   * @returns
+   */
+  async shareItem(
+    actor: Member,
+    repositories: Repositories,
+    itemId: Item['id'],
+    invitations: Pick<Invitation, 'email' | 'permission'>[],
+  ): Promise<{ memberships: ItemMembership[]; invitations: Invitation[] }> {
+    await this.itemService.get(actor, repositories, itemId, PermissionLevel.Admin);
+
+    if (!invitations.length) {
+      throw new NoInvitationReceivedFound();
+    }
```

Comment:
We should check if invitations is empty in the controller instead of in the service. It could prevent the `buildRepository` call and the creation of a SQL Transaction.

This is only a suggestion but if we consider the call of this function with empty invitations as bad, we can prevent it with a custom type (source : https://dev.to/martinpersson/type-safety-with-non-empty-arrays-in-typescript-k78)

```typescript
type NonEmptyArray<T> = [T, ...T[]]
function isNonEmptyArray<T>(arr: T[]): arr is NonEmptyArray<T>{
    return arr.length > 0
}

function func(_: NonEmptyArray<number>){

}

func([]) // WRONG
func([1]) // OK
func([1,2,3]) // OK

function sample(arr: number[]){
    func(arr); // WRONG
    if(arr.length>0){
        func(arr); // WRONG
    }
    if(isNonEmptyArray(arr)){
        func(arr); // OK
    }
}
```

```suggestion
  async shareItem(
    actor: Member,
    repositories: Repositories,
    itemId: Item['id'],
    invitations: NonEmptyArray<Pick<Invitation, 'email' | 'permission'>>,
  ): Promise<{ memberships: ItemMembership[]; invitations: Invitation[] }> {
    await this.itemService.get(actor, repositories, itemId, PermissionLevel.Admin);

```